gaphor/diagram/general/tests/test_generalpropertypages.py

```diff
@@ -42,3 +47,125 @@ def test_metadata_property_page(diagram, event_manager):
     description.set_text("my text")
 
     assert metadata.description == "my text"
+
+
+def test_picture_property_select_opens_dialog(mocker, diagram, event_manager):
+    # Init objects
+    picture = diagram.create(PictureItem)
+    property_page = PicturePropertyPage(picture, event_manager)
+    widget = property_page.construct()
+
+    # Prepare mocking
+    mocked_open_file_dialog = mocker.patch(
+        "gaphor.diagram.general.generalpropertypages.open_file_dialog"
+    )
+
+    # Emulate the button click event
+    button_widget = find(widget, "select-picture-button")
+    button_widget.set_activate_signal_from_name("clicked")
+
+    # Test code
+    button_widget.activate()
+
+    assert mocked_open_file_dialog.called is True
+
+
+def test_picture_property_select_valid_name(mocker, diagram, event_manager):
+    # Init objects
+    picture = next(
+        tl for tl in general_tools.tools if tl.id == "toolbox-picture"
+    ).item_factory(diagram)
+    property_page = PicturePropertyPage(picture, event_manager)
+
+    # Prepare mocking
+    mocked_error_handler = mocker.patch(
+        "gaphor.diagram.general.generalpropertypages.error_handler"
+    )
+
+    # Test code
+    temp_image = os.path.abspath("data/logos/gaphor-24x24.png")
+    property_page.open_file(temp_image)
+
+    assert mocked_error_handler.called is False
+    assert picture.subject.name == "gaphor-24x24"
+
+
+def test_picture_property_select_keep_name(mocker, diagram, event_manager):
+    # Init objects
+    picture = next(
+        tl for tl in general_tools.tools if tl.id == "toolbox-picture"
+    ).item_factory(diagram)
+    property_page = PicturePropertyPage(picture, event_manager)
+
+    # Prepare mocking
+    mocked_error_handler = mocker.patch(
+        "gaphor.diagram.general.generalpropertypages.error_handler"
+    )
+
+    # Test code
+    picture.subject.name = "old_name"
+    temp_image = os.path.abspath("data/logos/gaphor-24x24.png")
```

Comment:
It's more in line with the rest of the code to use `pathlib.Path` instead.

Reply:
done